src/ServiceManager.php

```diff
@@ -606,9 +606,10 @@ private function createDelegatorFromName($name, ?array $options = null)
 
             $this->delegators[$name][$index] = $delegatorFactory;
 
-            $creationCallback = function () use ($delegatorFactory, $name, $creationCallback, $options) {
-                return $delegatorFactory($this->creationContext, $name, $creationCallback, $options);
-            };
+            /**
+             * @psalm-suppress MissingClosureReturnType
+             */
+            $creationCallback = fn() => $delegatorFactory($this->creationContext, $name, $creationCallback, $options);
```

Comment:
I think this has been already adjusted in `3.9.x`

Reply:
rebased.